tests/tracer/telemetry/test_writer.py

```diff
@@ -0,0 +1,223 @@
+import json
+
+import httpretty
+import mock
+import pytest
+
+from ddtrace.internal.telemetry.data import get_application
+from ddtrace.internal.telemetry.data import get_host_info
+from ddtrace.internal.telemetry.writer import AGENT_URL
+from ddtrace.internal.telemetry.writer import ENDPOINT
+from ddtrace.internal.telemetry.writer import TelemetryWriter
+from ddtrace.settings import _config as config
+
+
+TEST_URL = "%s/%s" % (AGENT_URL, ENDPOINT)
+
+
+@pytest.fixture
+def mock_send_request_200():
+    with httpretty.enabled():
+        httpretty.register_uri(httpretty.POST, TEST_URL, status=202)
+        yield
+
+
+@pytest.fixture
+def telemetry_writer():
+    telemetry_writer = TelemetryWriter()
+    telemetry_writer.enabled = True
+    yield telemetry_writer
+
+
+@pytest.fixture
+def disabled_telemetry_writer():
+    # TelemetryWriter should be disabled by default
+    telemetry_writer = TelemetryWriter()
+    assert telemetry_writer.enabled is False
+
+    yield telemetry_writer
+
+
+def test_add_app_started_event(telemetry_writer):
+    """asserts that enabling the Telemetry writer queues and then sends an app-started telemetry request"""
+    telemetry_writer.app_started_event()
+
+    events = telemetry_writer._flush_events_queue()
+    assert len(events) == 1
+    assert events[0]["request_type"] == "app-started"
```

Comment:
The best way to test this class is by using the public API and then validating the http requests that are made.

For example, calling `telemetry_writer.app_started_event()` and then `telemetry_writer.shutdown()` and then validate that 1 HTTP call was made with the proper headers and request body.

And the same for all other event types.

You will also want to validate that add X integration events, calling periodic, add Y more integration events, and then calling shutdown will result in 2 HTTP calls.